### PR TITLE
Notification api changes

### PR DESCRIPTION
**Pull request by ranquild**
Support BE changes https://github.com/metabase/metabase/pull/17740:
- Always archive alerts and pulses instead of deleting them
- Use new `subscription` endpoint for unsubscribing

How to test:
- Merge the referenced PR into this branch locally
- Run cypress tests, e.g. `notifications.cy.spec.js`

### CODE REVIEW

**Comment by gusaiani** [comment]
@ranquild I had taken a look at this PR earlier today.

Now I see some file changes that seem quite unrelated to the gist of this PR: markdown about Percy etc.
Was this intentional?

File: frontend/src/metabase/alert/alert.js [hunk 1]
```diff
@@ -132,13 +132,13 @@ export const unsubscribeFromAlert = alert => {
...
 const deleteAlertRequest = new RestfulRequest({
-  endpoint: AlertApi.delete,
+  endpoint: AlertApi.update,
   actionPrefix: DELETE_ALERT,
   storeAsDictionary: true,
 });
 export const deleteAlert = alertId => {
   return async (dispatch, getState) => {
-    await dispatch(deleteAlertRequest.trigger({ id: alertId }));
+    await dispatch(deleteAlertRequest.trigger({ id: alertId, archived: true }));
```
Comment (kulyk) [inline frontend/src/metabase/alert/alert.js:141]: Should we maybe call it `ARCHIVE_ALERT` / `archiveAlertRequest` / `archiveAlert` in the code, so it's explicit what's actually going to happen to an alert?

**Review by kulyk — APPROVED** [review]
FE looks good, have just a minor naming suggestion
I suppose we should better rebase the branches, there are some extra commits coming from `master`

**Comment by kulyk** [comment]
> @ranquild I had taken a look at this PR earlier today.
> 
> Now I see some file changes that seem quite unrelated to the gist of this PR: markdown about Percy etc.
> Was this intentional?

Looks like this PR is targeting Noah's branch which is a few commits behind `master`, and this PR's branch is based on a more recent version of `master`. Rebasing this PR on Noah's branch or rebasing Noah's branch on the latest `master` should do the trick

CC @ranquild @noahmoss

**Comment by noahmoss** [comment]
I've rebased my branch which seems to have fixed the unrelated changes from `master`. There are still clojure changes from my PR duplicated in this PR, maybe this one needs to be rebased as well?

Anyway, @ranquild, whenever you're ready feel free to squash & merge this PR into my branch. I'll merge them both into `master` together after the BE is reviewed.

**Pull request merged**